src/providers/postgres/qgspostgresdataitemguiprovider.cpp

```diff
@@ -149,6 +149,31 @@ void QgsPostgresDataItemGuiProvider::populateContextMenu( QgsDataItem *item, QMe
       maintainMenu->addAction( actionRefreshMaterializedView );
     }
     menu->addMenu( maintainMenu );
+
+    if ( layerInfo.relKind != Qgis::PostgresRelKind::View && layerInfo.relKind != Qgis::PostgresRelKind::MaterializedView )
```

Comment:
I think this should be a list of accepted rel kinds, as it wouldn't work for an index or a sequence?

Reply:
good point, it can be even made more general by using views etc.